internal/pkg/remote/manager.go

```diff
@@ -165,18 +165,18 @@ func (u *UnitOfWork) SaveObject(objectState model.ObjectState, object model.Obje
 	}
 
 	// Invoke mapper
+	apiObject := object.Clone()
 	recipe := &model.RemoteSaveRecipe{
 		ChangedFields:  changedFields,
 		ObjectManifest: objectState.Manifest(),
-		InternalObject: object,
-		ApiObject:      object.Clone(),
+		Object:         apiObject,
 	}
 	if err := u.mapper.MapBeforeRemoteSave(recipe); err != nil {
 		u.errors.Append(err)
 		return
 	}
 
-	if err := u.createOrUpdate(objectState, recipe, changedFields); err != nil {
+	if err := u.createOrUpdate(objectState, object, recipe, changedFields); err != nil {
```

Comment:
Pri `save` musime pridat dalsi parameter `object`, ... to je interna reprezentacia ... ktora sa po uspesnom ulozeni ulozi cez `SetRemoteState`.